tests/python_package_test/test_dask.py

```diff
@@ -273,7 +276,7 @@ def test_classifier(output, task, boosting_type, client):
     p2_proba = local_classifier.predict_proba(X)
     s2 = local_classifier.score(X, y)
 
-    if boosting_type == 'rf' and output == 'dataframe-with-categorical':
```

Comment:
IS this change related to this PR?

Reply:
Yes. Because of https://github.com/microsoft/LightGBM/issues/4118, rf for classification isn't very reliable. When using `tree_learner='data'` these tests would pass, but with voting they all fail, so I moved all of the rf tests to the "soft check" instead of just the ones for `dataframe-with-categorical`.

Reply:
Got it, thanks for the explanation!